src/commands/hld/reconcile.ts

```diff
@@ -309,25 +311,51 @@ export const reconcileHld = async (
         continue;
       }
 
+      // Calculate shared path for both IngressRoute and Middleware
+      const ingressVersionAndPath = getFullPathPrefix(
+        serviceConfig.pathPrefixMajorVersion || "",
+        serviceConfig.pathPrefix || "",
+        serviceName
+      );
+
       // Create middleware.
       const middlewares = dependencies.createMiddlewareForRing(
         ringPathInHld,
-        serviceName,
-        ring
+        serviceConfig.displayName || serviceName,
+        ring,
+        ingressVersionAndPath
       );
 
       // Create Ingress Route.
       dependencies.createIngressRouteForRing(
         ringPathInHld,
-        serviceName,
+        serviceConfig.displayName || serviceName,
         serviceConfig,
         middlewares,
-        ring
+        ring,
+        ingressVersionAndPath
       );
     }
   }
 };
 
+/**
+ * Build and return the full path prefix used for IngressRoutes and Middlewares
+ * @param majorVersion
+ * @param pathPrefix
+ * @param serviceName
+ */
+export const getFullPathPrefix = (
+  majorVersion: string,
+  pathPrefix: string,
+  serviceName: string
+): string => {
+  const versionPath = majorVersion ? `/${majorVersion}` : "/";
+  const servicePath = pathPrefix || serviceName;
+
+  return path.join(versionPath, servicePath);
```

Comment:
I thought this should be
```
  const versionPath = majorVersion ? `/${majorVersion}` : "";
  const servicePath = pathPrefix || serviceName;
  return `${versionPath}/${servicePath}`;
```
this is URL path and not file system path (if i am not wrong)

Reply:
@dennisseah I think @mtarng's fix should be fine - we're specifying the full path to strip (from the root) of the url ie:

http://bar.com/v1/fabrikam should strip /v1/fabrikam

Reply:
`path` library should not be used on constructing URL. separator is always `/` if there a reason for using `path`? Thanks

Reply:
https://stackoverflow.com/questions/16301503/can-i-use-requirepath-join-to-safely-concatenate-urls